frontend/packages/console-app/src/components/vpa/VerticalPodAutoscalerRecommendations.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const FilterVerticalPodAutoscaler = (vpas, obj) =>
+  (vpas ?? []).find((vpa) => {
+    const { targetRef } = vpa.spec;
+    return (
+      targetRef &&
+      targetRef.apiVersion === obj?.apiVersion &&
+      targetRef.kind === obj?.kind &&
+      targetRef.name === obj?.metadata?.name
+    );
+  });
+
+export const VerticalPodAutoscalerRecommendations: React.FC<VerticalPodAutoscalerRecommendationsProps> = ({
+  obj,
+}) => {
+  const { t } = useTranslation();
+  const [vpas] = useK8sWatchResource<K8sResourceCommon[]>({
+    groupVersionKind: {
```

Comment:
I would suggest to pass the object's GVK, so we dont need to do all the matching in the `FilterVerticalPodAutoscaler` function, we only check for name.
```
groupVersion = groupVersionFor(obj.apiVersion)
{
      group: groupVersion.group,
      version: groupVersion.version,
      kind: obj.kind,
},
```